### DISCUSSION TITLE
Crossreferencing/Custom floats for Quarto websites

### DISCUSSION THREAD

**Discussion by PierreENOlivier** [quarto-dev/quarto-cli#9004] [Q&A] [answered]
### Description

Hello,
I was looking at the documentation and stumble upon "[Crossreferences and Custom Floats](https://quarto.org/docs/authoring/cross-references-custom.html)".

Any idea if it will be added to Quarto Websites?

The way I would see things work would be to reference from one page to another.
For instance, you could define a quarto section (not sure the name but the ones defines with `:::`) in a `.qmd` file and reference it anywhere you need.

For instance, I could embed a subscription form and reference that code anywhere I need.
If I need to modify that code, I would only need to change the code in one location and re-render the website.

In an email.qmd
```
:::{#code-email}
<some-embed></>
:::
```

Anywhere else I need:
`@code-email`

A footer can do the trick, but it will always be at the same location: the bottom.
With crossrefs, I could decide if I want to insert the code in the middle of my posts, or anywhere else.

Best,

**Comment by mcanouil** [comment, marked as answer]
Did you try? It should work across formats.
Since it works for HTML, there is no reason for it to not work on Books/Websites.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/4f671ffd-fc25-486a-94d8-47e7d7ff40cf">
